### PR TITLE
feat(html pipe) - URL matching from markup.yaml

### LINKED ISSUE #688: URL matching from markup.yaml
With https://github.com/adobe/helix-shared/pull/280 in the books, we can start on https://github.com/adobe/helix-shared/issues/279

### PR DESCRIPTION
**Pull request by ramboz**
Add support for matching against URL and adjusting MDAST accordingly
Requires: #687

fix #688

### CODE REVIEW

File: src/utils/adjust-markup.js [hunk 1]
```diff
@@ -66,6 +68,39 @@ function getHTMLElement(template) {
   return dom.firstChild;
 }
 
+/**
```
Comment (ramboz) [inline src/utils/adjust-markup.js:71]: This is coming from #687

File: src/utils/adjust-markup.js [hunk 2]
```diff
@@ -85,36 +120,32 @@ async function adjustMDAST(context, action) {
...
+    });
+
   Object.entries(markupconfig.markup)
     .filter(([_, cfg]) => cfg.type === 'markdown')
     .forEach(([name, cfg]) => {
       logger.info(`Applying markdown markup adjustment: ${name}`);
 
       transformer.match(cfg.match, (h, node) => {
         const handler = transformer.constructor.default(node);
-        let el = handler(h, node);
```
Comment (ramboz) [inline src/utils/adjust-markup.js:95]: This is coming from #687

File: src/utils/adjust-markup.js [hunk 3]
```diff
@@ -85,36 +120,32 @@ async function adjustMDAST(context, action) {
     return;
   }
 
+  const sectionHandler = section();
```
Comment (ramboz) [inline src/utils/adjust-markup.js:123]: This is new

File: test/testPipeWithMarkupConfig.js [hunk 4]
```diff
@@ -81,6 +96,46 @@ describe('Testing HTML Pipeline with markup config', () => {
       secrets: {},
       logger,
     });
+    nock.restore();
+    nock.activate();
+    nock.cleanAll();
+  });
+
+  it('html.pipe adjusts the MDAST as per markup url config', async () => {
```
Comment (ramboz) [inline test/testPipeWithMarkupConfig.js:104]: This is new, rest is coming from #687

**Comment by trieloff** [comment]
So excited to have this feature. Thanks for your patience and persistence, @ramboz.

**Comment by rofe** [comment]
> So excited to have this feature.

Same here! We should try it out on theblog in order to get rid of [this](https://github.com/adobe/theblog/blob/34da4c93fc1b6ba2f08ea5d935f8ebbad4c41a31/head.html#L43-L69) and [this](https://github.com/adobe/theblog/blob/34da4c93fc1b6ba2f08ea5d935f8ebbad4c41a31/scripts.js#L199).

**Pull request merged**